pages/api/needle.js

```diff
@@ -28,7 +28,7 @@ export default async function handler(request, response) {
   if (request.method === "PUT") {
     try {
       const { id } = request.query;
-      await Yarn.findByIdAndUpdate(id, { $set: request.body });
+      await Needle.findByIdAndUpdate(id, { $set: request.body });
       response.status(201).json({ status: "yarn successfully updated" });
```

Comment:
To be more precise, better deliver statusMessage or just `message` as key instead of `status` in your responses: 
```suggestion
      response.status(201).json({ message: "yarn successfully updated" });
```